Sources/SessionManager/SessionManager+Extension.swift

```diff
@@ -128,7 +128,20 @@ extension SessionManager {
         let newXValue = reversedSharedSecret.hexa
         let hash = SHA2(variant: .sha512).calculate(for: newXValue.hexa).hexa
         let aesEncryptionKey = hash.prefix(64)
+        
+        let _sharedSecretPrefix = Array(tupleToArray(sharedSecretData).prefix(32))
+        let _reversedSharedSecret = _sharedSecretPrefix.uint8Reverse()
+        let _hash = SHA2(variant: .sha512).calculate(for: Array(_reversedSharedSecret))
+        let macKey = Array(_hash.suffix(32))
```

Comment:
why repeat the code?
why cant use suffix(32) no the hash value?

Reply:
this code is not repeated. Hash is being derived different way for hmac verification.